### PR TITLE
HIP-584: Adapt PrecompilePricingUtils to work with fee subtypes

### LINKED ISSUE #6366: HIP-584: Adapt `PrecompilePricingUtils` to work with fee subtypes
### Problem

Currently `PrecompilePricingUtils` works only for `DEFAULT` fee subtype. This causes some precompiles like mint and wipe to fail, since they rely on using `FUNGIBLE_TOKEN` and `NON_FUNGIBLE_TOKEN` subtypes.

### Solution

Adapt `PrecompilePricingUtils` to work with all fee subtypes.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:  

Copy missing Pricing logic from hedera-services 

**Related issue(s)**:

Fixes #6366

**Notes for reviewer**:
  

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/utils/accessors/SignedTxnAccessor.java [hunk 1]
```diff
@@ -16,33 +16,69 @@
...
+    private BaseTransactionMeta txnUsageMeta;
+    private static final ExpandHandleSpanMapAccessor SPAN_MAP_ACCESSOR = new ExpandHandleSpanMapAccessor();
     private final int sigMapSize;
     private final int numSigPairs;
     private final byte[] hash;
     private final byte[] txnBytes;
+    private final byte[] utf8MemoBytes;
     private final String memo;
+
+    private final TokenWipeAccountTransactionBody body;
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/utils/accessors/SignedTxnAccessor.java:81]: I believe the type should be `TransactionBody`

Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/utils/accessors/SignedTxnAccessor.java]: Fixed

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java [hunk 2]
```diff
@@ -699,6 +739,25 @@ protected void exchangeRatesPersist() {
     }
 
     protected void feeSchedulesPersist() {
+        //        for (var function: HederaFunctionality.values()) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java:742]: Adapt code or remove comments

File: hedera-mirror-web3/src/test/java/com/hedera/services/fees/calculation/utils/PricedUsageCalculatorTest.java [hunk 3]
```diff
@@ -94,30 +105,74 @@ void delegatesSupports() {
...
+    //
+    //        given(calculator.fees(feesCaptor.capture(), eq(mockPrices), eq(mockRate), longThat(l -> l == multiplier)))
+    //                .willReturn(mockFees);
+    //
+    //        // when:
+    //        final var actual = subject.extraHandleFees(mockPrices, mockRate);
+    //
+    //        // then:
+    //        assertEquals(mockFees, actual);
+    //    }
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/services/fees/calculation/utils/PricedUsageCalculatorTest.java:177]: Fix tests

File: hedera-mirror-web3/src/main/java/com/hedera/services/utils/accessors/SignedTxnAccessor.java [hunk 4]
```diff
@@ -100,9 +137,13 @@ protected SignedTxnAccessor(final byte[] signedTxnWrapperBytes, @Nullable final
         txnId = txn.getTransactionID();
         sigMapSize = sigMap.getSerializedSize();
         numSigPairs = sigMap.getSigPairCount();
+        utf8MemoBytes = StringUtils.getBytesUtf8(memo);
         payer = getTxnId().getAccountID();
 
         getFunction();
+        setBaseUsageMeta();
+        setOpUsageMeta();
+        setTokenWipeUsageMeta();
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/utils/accessors/SignedTxnAccessor.java:146]: Why do we need explicit setTokenWipeUsageMeta(), when we don't have this method in services?

File: hedera-mirror-web3/src/test/java/com/hedera/services/fees/calculation/utils/PricedUsageCalculatorTest.java (hunk 3 above)
Comment (0xivanov) [reply hedera-mirror-web3/src/test/java/com/hedera/services/fees/calculation/utils/PricedUsageCalculatorTest.java]: fixed

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**